cherab/solps/formats/balance.py

```diff
@@ -0,0 +1,192 @@
+    ##########################################
+
+    sim._species_list = []
+    n_species = len(fhandle.variables['am'].data)
+	
+    for i in range(n_species):
+
+        # Extract the nuclear charge	
+        if fhandle.variables['species'].data[i,1] == b'D':
+	        zn = 1
+        if fhandle.variables['species'].data[i,1] == b'C':
+            zn = 6
+        if fhandle.variables['species'].data[i,1] == b'N':
+            zn = 7
+        if fhandle.variables['species'].data[i,1] == b'N' and fhandle.variables['species'].data[i,2] == b'e':
+            zn = 10
+        if fhandle.variables['species'].data[i,1] == b'A' and fhandle.variables['species'].data[i,2] == b'r':
+            zn = 18
+	
+        am = float(fhandle.variables['am'].data[i])  # Atomic mass
+        charge = int(fhandle.variables['za'].data[i])  # Ionisation/charge
+        species = _popular_species[(zn, am)]
+		
+        # If we only need to populate species_list, there is probably a faster way to do this...		
+        sim.species_list.append(species.symbol + str(charge))
+
+    tmp = copy.deepcopy(fhandle.variables['na'].data)
+    tmp = swapaxes(tmp,0,2)
+    tmp = swapaxes(tmp,0,1)
+    sim._species_density = tmp
+
+    # Make Mesh Interpolator function for inside/outside mesh test.
+    inside_outside_data = ones(mesh.num_tris)
+    inside_outside = AxisymmetricMapper(Discrete2DMesh(mesh.vertex_coords, mesh.triangles, inside_outside_data, limit=False))
+    sim._inside_mesh = inside_outside
+	
+    # Load the neutrals data
+    if 'D0' in sim.species_list:
+        for i in arange(len(sim.species_list)):
+            if sim.species_list[i] == 'D0':
+                D0_indx = i
+
+    # Replace the deuterium neutrals density (from the fluid neutrals model by default) with
+    # the values calculated by EIRENE - do the same for other neutrals?
+    if 'dab2' in fhandle.variables.keys():
+        sim.species_density[:,:,D0_indx] = copy.deepcopy(fhandle.variables['dab2'].data[0,:,0:-2])
+        eirene_run = True
+    else:
+        eirene_run = False
+    
+    # Calculate the total radiated power
+    
+    
+    if eirene_run is True:
+        # Total radiated power from B2, not including neutrals
+        b2_ploss = fhandle.variables['b2stel_she_bal'].data/vol
+        
+        # Electron energy loss due to interactions with neutrals
+        if 'eirene_mc_eael_she_bal' in fhandle.variables.keys():
+            eirene_ecoolrate = sum(fhandle.variables['eirene_mc_eael_she_bal'].data,axis=0)/vol
+            
+        # Ionisation rate from EIRENE, needed to calculate the energy loss to overcome the ionisation potential of atoms
+        if 'eirene_mc_papl_sna_bal' in fhandle.variables.keys():
+            eirene_potential_loss = 13.6*sum(fhandle.variables['eirene_mc_papl_sna_bal'].data,axis=(0))[1,:,:]*Q/vol
+        
+        # This will be negative (energy sink); take absolute valu
+        sim._total_rad = abs(b2_ploss+(eirene_ecoolrate-eirene_potential_loss))
```

Comment:
Is it the absolute value we want here? Or do we strictly want to return the negative of the energy loss for the radiated power? Semantically I think the latter, but is there ever the possibility of the energy loss term returned by Eirene being positive (which would correspond to negative radiated power)?

Reply:
Done, the term in brackets is now multiplied by -1